test/parallel/test-buffer.js

```diff
@@ -609,6 +609,15 @@ for (var i = 0; i < 256; i++) {
   assert.equal(hexb2[i], hexb[i]);
 }
 
+// invalid HEX strings should throw an error
+assert.throws(() => new Buffer('xx', 'hex'), /TypeError: Invalid hex string/);
+// following will fail even though it has zero as the first character, because
+// `x` is an invalid hexa-decimal value
+assert.throws(() => new Buffer('0x', 'hex'), /TypeError: Invalid hex string/);
+// following invalid HEX strings will fail because of the odd length
+assert.throws(() => new Buffer('0', 'hex'), /TypeError: Invalid hex string/);
+assert.throws(() => new Buffer('000', 'hex'), /TypeError: Invalid hex string/);
```

Comment:
Could you also add some test without the zeros or `x`-ses? `0x` could be treated as in incomplete prefix, which it isn't.
I suggest adding `FFABCT` and `ffcdabqa`, for example (different length, different invalid position (even/odd), different letter.